### PR TITLE
Setup for integration tests

### PR DESCRIPTION
**Pull request by axiomatic-aardvark**
### Description
This PR demonstrates a way to write integration tests for radios, using Wiremock and a bash script. 

The bash script does a few things:
1. Spins up a test boot node (process sent to background) that the light nodes will connect with.
2. Spins up N number of test light nodes(processes sent to background) to send messages to each other (currently set to 3)
3. Spins up one last light node in the foreground (logs are shown in the terminal that the test script was started from) that checks if it has received 4 messages and when it has - it stops, successfully demonstrating that the nodes are working properly and that messages are propagating and being handled by the `attestation_handler` function.

To do this, the bash script runs the `regression_test` test (which is a normal async Rust test, so we get the full benefit of the cargo testing tools). The test itself demonstrates how to use [wiremock-rs](https://github.com/LukeMathWalker/wiremock-rs) in order to mock the registry and network subgraphs (it can be used to mock anything that a Radio might need from a third party API).

The `regression_test` itself is rather simple, because after the messages are validated and handled, they're passed over to the `process_messages` function, which is already thoroughly tested by the existing unit tests.

**To run the integration test, use this command (after making the script executable):**
```
./run-tests.sh
```

### Issue link (if applicable)
Will close https://github.com/graphops/graphcast-sdk/issues/32

### Possible improvements and next steps
- This is just the ground work for integration testing, a lot more test cases can be written for the POI Radio, and also more generic examples can be developed for documentation and tutorial purposes.
- Currently, the tests use `test-poi-crosschecker-radio` as a Radio name for the test instances, but if two developers are testing the same Radio at the same time that will include noise. I'm leaving it for a future improvement, one way to solve it would be to generate a random UUID or something and append it to the Radio name, in order to guarantee that the test instances are not receiving any outside data.
- There can be a better way to run the integration tests, such as a custom cargo command that runs the bash script.
- The test is marked as with `#[ignore]` for now because I thought it would be best to separate it from the unit tests, also it doesn't run properly when using `nextest`, but in the future we could combine them though.

PS This PR still uses the (forked SDK repo)[https://github.com/axiomatic-aardvark/graphcast-sdk] as a dependency. Once [this PR](https://github.com/graphops/graphcast-sdk/pull/39) is merged, we can merge this one.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [package]
-name = "poi-crosschecker"
+name = "poi-radio"
 version = "0.1.0"
 edition = "2021"
 
 [dependencies]
-graphcast_sdk = { package = "graphcast-sdk", git = "https://github.com/graphops/graphcast-sdk", rev="ee94148" }
+graphcast_sdk = { package = "graphcast-sdk", git = "https://github.com/axiomatic-aardvark/graphcast-sdk" }
```
Comment (hopeyen) [inline Cargo.toml:7]: iirc you can reference with actual repo commit with 'rev' even if the commit hasn't been merging? Is there something preventing that development process such that a fork is needed?

File: src/graphql/mod.rs [hunk 2]
```diff
@@ -48,7 +48,7 @@ pub async fn query_graph_node_poi(
         match data.proof_of_indexing {
             Some(poi) => Ok(poi),
             // _ => Err(QueryError::EmptyResponseError),
```
Comment (hopeyen) [inline src/graphql/mod.rs:50]: 😂I might've accidentally committed this, but this line definitely should not have been commented. Could you please uncomment the EmptyResponseError and then comment or delete the workaround on line#51?

File: src/main.rs [hunk 3]
```diff
@@ -39,11 +40,18 @@ async fn main() {
     let wait_block_duration = 2;
 
     let provider: Provider<Http> = Provider::<Http>::try_from(eth_node.clone()).unwrap();
-    let radio_name: &str = "poi-crosschecker";
+    let radio_name: &str = "poi-crosschecker-radio";
```
Comment (hopeyen) [inline src/main.rs:43]: perhaps we should update 'poi-crosschecker-radio' references to just 'poi-radio'?

**Review by hopeyen — APPROVED** [review]
nice work on the set up! Let's create an issue for a more complete test suite such as first time sender and attestation conflicts:)

File: Cargo.toml (hunk 1 above)
Comment (axiomatic-aardvark) [reply Cargo.toml:7]: You're right, I was trying out a bunch of experimental stuff before that and that's why I had a fork, but I'll look into how to use a local git dependency in the future, that'd be easiest. Also yeah if a commit has been pushed (but not merged to main) we can still reference it, so no need of forks really

File: src/graphql/mod.rs (hunk 2 above)
Comment (axiomatic-aardvark) [reply src/graphql/mod.rs:50]: done ✅

File: src/main.rs (hunk 3 above)
Comment (axiomatic-aardvark) [reply src/main.rs:43]: also done ✅

**Review by hopeyen — DISMISSED** [review]

File: Cargo.toml (hunk 1 above)
Comment (hopeyen) [reply Cargo.toml:7]: reminder to the link before you merge, thanks!

**Review by hopeyen — APPROVED** [review]

**Pull request merged**